cookbooks/calico/recipes/control.rb

```diff
@@ -460,6 +481,19 @@
     action [:nothing]
 end
 
+# Output and store the UID and GID for nova - this may be required for live migration
+execute "get-nova-info" do
+    command "id nova >> /tmp/nova.user"
+end
+ruby_block "store-nova-user-info" do
+    block do
+        output = ::File.read("/tmp/nova.user")	
+        match = /uid=(?<uid>\d+).*gid=(?<gid>\d+).*/.match(output)
+	node.set["nova_uid"] = match[:uid]
```

Comment:
Ack, actually lots of lines do; maybe your editor doesn't make them into spaces for Ruby?